framework/src/actions/SetupRecoverFileBaseAction.C

```diff
@@ -49,14 +49,19 @@ SetupRecoverFileBaseAction::act()
     // Set the recover file base in the App
     mooseInfo("Using ", _app.getRestartRecoverFileBase(), " for recovery.");
   }
-  else // recover_mesh_meta_data
+  else // recover_meta_data
   {
     // Make sure that all of the mesh meta-data attributes have been declared (after the mesh
     // generators have run.
-
     RestartableDataIO restartable(_app);
-
-    if (restartable.readRestartableDataHeader(false, MeshMetaDataInterface::FILE_SUFFIX))
-      restartable.readRestartableData(_app.getMeshMetaData(), DataNames());
+    for (auto map_iter = _app.getRestartableDataMapBegin();
+         map_iter != _app.getRestartableDataMapEnd();
+         ++map_iter)
+    {
+      const RestartableDataMap & meta_data = map_iter->second.first;
+      const std::string & suffix = map_iter->second.second;
```

Comment:
Ah - so the other part of your data structure is the file suffix. You might want to mention that in MooseApp.h where you declare it. It wasn't clear.

Reply:
I can do that